ipc/logger/pipeLogger.go

```diff
@@ -0,0 +1,171 @@
+package logger
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"io"
+	"os"
+)
+
+var _ Logger = (*PipeLogger)(nil)
+
+// LogMessage is a log message
+type LogMessage struct {
+	Level   LogLevel
+	Message string
+	Args    []interface{}
+}
+
+// PipeLogger is a pipe-based logger
+type PipeLogger struct {
+	pipe     *os.File
+	level    LogLevel
+	fallback Logger
+}
+
+// NewPipeLogger creates a new pipe logger
+func NewPipeLogger(level LogLevel, pipe *os.File) Logger {
+	return &PipeLogger{
+		level:    level,
+		pipe:     pipe,
+		fallback: NewDefaultLogger(level),
+	}
+}
+
+// Trace logs
+func (pipeLogger *PipeLogger) Trace(message string, args ...interface{}) {
+	if pipeLogger.level > LogTrace {
+		return
+	}
+
+	pipeLogger.sendMessage(&LogMessage{
+		Level:   LogTrace,
+		Message: message,
+		Args:    args,
+	})
+}
+
+// Debug logs
+func (pipeLogger *PipeLogger) Debug(message string, args ...interface{}) {
+	if pipeLogger.level > LogDebug {
+		return
+	}
+
+	pipeLogger.sendMessage(&LogMessage{
+		Level:   LogDebug,
+		Message: message,
+		Args:    args,
+	})
+}
+
+// Info logs
+func (pipeLogger *PipeLogger) Info(message string, args ...interface{}) {
+	if pipeLogger.level > LogInfo {
+		return
+	}
+
+	pipeLogger.sendMessage(&LogMessage{
+		Level:   LogInfo,
+		Message: message,
+		Args:    args,
+	})
+}
+
+// Warn logs
+func (pipeLogger *PipeLogger) Warn(message string, args ...interface{}) {
+	if pipeLogger.level > LogWarning {
+		return
+	}
+
+	pipeLogger.sendMessage(&LogMessage{
+		Level:   LogWarning,
+		Message: message,
+		Args:    args,
+	})
+
+}
+
+// Error logs
+func (pipeLogger *PipeLogger) Error(message string, args ...interface{}) {
+	if pipeLogger.level > LogError {
+		return
+	}
+
+	pipeLogger.sendMessage(&LogMessage{
+		Level:   LogError,
+		Message: message,
+		Args:    args,
+	})
+}
+
+func (pipeLogger *PipeLogger) sendMessage(message *LogMessage) {
+	payload, err := marshalJSON(message)
+	if err != nil {
+		pipeLogger.fallback.Error("pipeLogger.sendMessage() marshal error", err.Error())
+		return
+	}
+
+	// Send length
+	length := len(payload)
+	buffer := make([]byte, 4)
+	binary.LittleEndian.PutUint32(buffer, uint32(length))
+	_, err = pipeLogger.pipe.Write(buffer)
+	if err != nil {
+		pipeLogger.fallback.Error("pipeLogger.sendMessage() send length error", err.Error())
+		return
+	}
+
+	// Send payload
```

Comment:
no need for comments inside the function.

Reply:
Fixed.